test/testother.cpp

```diff
@@ -7066,6 +7066,14 @@ class TestOther : public TestFixture {
               "    if ($a == $a) { }\n"
               "}");
         ASSERT_EQUALS("", errout_str());
+
+        checkP("#define X 1\n"
```

Comment:
I'm not sure what macroName there will be if there are nested macros. Can you try this:

    #define X 1
    #define Y X
    void f() {
         if (X == Y) {}
    }

I think that ideally we don't warn about that because Y could possibly have different definitions..

Reply:
it strikes me that it might be within reach to identify if a macro can have different possible values in different configurations.. if the code is something like:

    #ifdef __GNUC__
    #define X 1
    #else
    #define X 2
    #endif

I think it would not be impossible to mark "X" as a macro that has different values using the info from simplecpp that I think is available. But the info must be propagated to the checkers somehow..

Reply:
We also don't know if a macro was a simple define or a functional-style macro, which should be handled differently (see the failing test in windows.cpp).